app/models/minion.rb

```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
 # Minion represents the minions that have been registered in this application.
 class Minion < ApplicationRecord
+  validates :hostname, uniqueness: true
```

Comment:
It shouldn't. Just taking one extra precaution to avoid invalid data in the database.